### PR TITLE
#115900 - TeamsV2: Graph Connector doesn't stop syncing when the MSFT…

### PR DESCRIPTION
**Pull request by GitNedim**
… tenant quota has been exceeded

PR contains fix where we need to stop sync if Tenant quota limit has excided. 

We have 2 cases when we need to stop sync and display error:
1. When application is started and we get connection which Tenant quota limit is already exceeded. (Initialize method)
2. While syncing items Tenant quota limit exceeded. 

Tested both cases using **panoptodev4** tenant. 

**Resolved Workitem(s):** 115900

### CODE REVIEW

File: src/panoptoindexconnector/connector.py [hunk 1]
```diff
@@ -334,6 +335,8 @@ def sync(config, last_update_time):
     except requests.exceptions.HTTPError as ex:
         LOG.exception('Received error response %s | %s', ex.response.status_code, ex.response.text)
         exception = ex
+    except CustomExceptions.QuotaLimitExceededError as ex:
```
Comment (hohno-panopto) [inline src/panoptoindexconnector/connector.py:338]: Don't we need to log here, as we do for other error cases?

File: src/panoptoindexconnector/implementations/microsoft_graph_implementation.py [hunk 2]
```diff
@@ -85,6 +88,16 @@ def push_to_target(target_content, config):
...
+    # If request is forbidden
+    elif response.status_code == 403:
+        error = response.json()["error"]
+
+        if error and error.get("innerError"):
+            innerError = error.get("innerError")
+            if innerError.get('code') == "TenantQuotaExceeded":
+                raise CustomExceptions.QuotaLimitExceededError(innerError.get("message"))
+
+        response.raise_for_status()
```
Comment (hohno-panopto) [inline src/panoptoindexconnector/implementations/microsoft_graph_implementation.py]: Is this line a residual of development? Also, isn't it true that your new code now loses a log for the case that the response is 403 but not from quota issue? Before your change, it went to else clause and it has a log line.

**Review by hohno-panopto — COMMENTED** [review]
See comments inline. Grab me if in-person discussion is faster.

File: src/panoptoindexconnector/implementations/microsoft_graph_implementation.py (hunk 2 above)
Comment (GitNedim) [reply src/panoptoindexconnector/implementations/microsoft_graph_implementation.py]: You are right. Added missing log. Thanks for catching!

File: src/panoptoindexconnector/connector.py (hunk 1 above)
Comment (GitNedim) [reply src/panoptoindexconnector/connector.py:338]: No need to log since it will be logged in caller method. I have added a comment for that. Thanks.

**Review by hohno-panopto — APPROVED** [review]

**Pull request merged**